mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/NoopAttachmentMapper.java

```diff
@@ -26,6 +26,7 @@
 import org.apache.james.mailbox.exception.MailboxException;
```

Comment:
IMO this NoopAttachmentMapper is dangerous.

We should remove it.

The way to do it:

 - **MailboxSessionMapperFactory** should not implement **AttachmentMapperFactory**
 - Cassandra and memory factories should implement **AttachmentMapperFactory**
 - Unbind **AttachmentMapperFactory** in JPAMailboxModule
 - extract attachment storage from **StoreMessageManager** **appendMessageToStore** line 637-643  in *a protected empty method*. This can the be overriden in **InMemoryMessageManager** and **CassandraMessageManager** to actually store attachment.

We would gain **compile time safety** and not let potential developer rely on an AttachmentMapper in an implementation that does not support it.

Reply:
done